src/lidar_measurement_model_beam.cpp

```diff
@@ -98,6 +98,9 @@ void LidarMeasurementModelBeam::loadConfig(
   filter_label_max_ = filter_label_max;
 
   pnh.param("add_penalty_short_only_mode", add_penalty_short_only_mode_, true);
+  double hit_range;
+  pnh.param("hit_range", hit_range, 0.3);
```

Comment:
Please reverse the number sign of hit_range when `add_penalty_short_only_mode_` is false, as the ray must be extended.

Reply:
I think this logic is wrong.
I added a visualization of likelihood function and beam status during ray casting.

```
                                                                      ↓ wall at x=2
|        -4        -3        -2        -1         0         1         2         3         4         
|         |         |         |         |         0         |         |         |         |         
_________________________________________________▇▇▇________________________________________________
likelihood is         low                       high                        low

llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll**ssssssssssssssssssssssssssss
beam status is                        long                            hit             short
```

```
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 0
|         |         |         |         |         0         |         |         |         |         
____________________________________________________________________________________________________
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllssssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 0.2
|         |         |         |         |         0         |         |         |         |         
_________________________________________________▇▇▇________________________________________________
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll**ssssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 0.4
|         |         |         |         |         0         |         |         |         |         
_______________________________________________▇▇▇▇▇▇▇______________________________________________
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*******ssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 0.6
|         |         |         |         |         0         |         |         |         |         
______________________________________________▇▇▇▇▇▇▇▇▇▇____________________________________________
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***********sssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 0.8
|         |         |         |         |         0         |         |         |         |         
____________________________________________▇▇▇▇▇▇▇▇▇▇▇▇▇▇__________________________________________
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***************sssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 1
|         |         |         |         |         0         |         |         |         |         
____________________________________________▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇________________________________________
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll******************ssssssssssssssssssss
```

In `add_penalty_short_only_mode=1`, hit is treated as short.
```
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 0
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇_________▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllssssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 0.2
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇_______▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 0.4
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇_____▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 0.6
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇___▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 0.8
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇_▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllsssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 1
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllssssssssssssssssss
```
Another problem: short is appeared at long side in DDA.
```
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 0
|         |         |         |         |         0         |         |         |         |         
____________________________________________________________________________________________________
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllsssssssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 0.2
|         |         |         |         |         0         |         |         |         |         
__________________________________________________▇▇________________________________________________
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllss***ssssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 0.4
|         |         |         |         |         0         |         |         |         |         
__________________________________________________▇▇▇▇______________________________________________
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllss*******ssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 0.6
|         |         |         |         |         0         |         |         |         |         
__________________________________________________▇▇▇▇▇▇____________________________________________
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllss***********ssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 0.8
|         |         |         |         |         0         |         |         |         |         
__________________________________________________▇▇▇▇▇▇▇▇__________________________________________
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllss***************ssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 1
|         |         |         |         |         0         |         |         |         |         
__________________________________________________▇▇▇▇▇▇▇▇▇▇________________________________________
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllss*******************ssssssssssssssssssss
```
```
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 0
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇____________▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllsssssssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 0.2
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇_________▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*ssssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 0.4
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇_______▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*ssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 0.6
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇_____▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*ssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 0.8
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇___▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*ssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 1
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇_▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*ssssssssssssssssssss
```

Reply:
Fixed version in this branch
```
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 0
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllssssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 0.2
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▇▇▇▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll**ssssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 0.4
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▇▇▇▇▇▇▇▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*******ssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 0.6
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▇▇▇▇▇▇▇▇▇▇▇▃▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***********sssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 0.8
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▆▇▇▇▇▇▇▇▇▇▇▇▇▇▇▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***************sssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 0, hit_range: 1
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll******************ssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 0
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▁▁▁▁▁▁▁▁▁▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllssssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 0.2
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▁▁▁▁▁▁▁▁▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll**ssssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 0.4
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▁▁▁▁▁▁▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*******ssssssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 0.6
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▃▁▁▁▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***********sssssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 0.8
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▁▁▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***************sssssssssssssssssssss
use_raycast_using_dda: 0, add_penalty_short_only_mode: 1, hit_range: 1
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll******************ssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 0
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllsssssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 0.2
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▃▇▇▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***ssssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 0.4
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▂▂▃▇▇▇▇▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*******ssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 0.6
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▂▂▂▂▃▇▇▇▇▇▇▃▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***********ssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 0.8
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▂▂▂▂▂▂▃▇▇▇▇▇▇▇▇▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***************ssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 0, hit_range: 1
|         |         |         |         |         0         |         |         |         |         
▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▂▂▂▂▂▂▂▂▃▇▇▇▇▇▇▇▇▇▇▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*******************ssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 0
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▅▁▁▁▁▁▁▁▁▁▃▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllsssssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 0.2
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▁▁▁▁▁▁▁▁▃▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***ssssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 0.4
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▁▁▁▁▁▁▃▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*******ssssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 0.6
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▃▁▁▁▃▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***********ssssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 0.8
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▂▁▃▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll***************ssssssssssssssssssssss
use_raycast_using_dda: 1, add_penalty_short_only_mode: 1, hit_range: 1
|         |         |         |         |         0         |         |         |         |         
▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▃▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇
lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll*******************ssssssssssssssssssss
```

(`use_raycast_using_dda: true`, `add_penalty_short_only_mode: false` looks strange though)